src/app/portlets/content-types/edit/content-types-edit.component.spec.ts

```diff
@@ -39,85 +46,167 @@ class TestContentTypeLayoutComponent {
+        expect(cancelButton === null).toBe(false);
+    });
+
+    it('should have save button and disabled by default', () => {
+        const saveButton = de.query(By.css('.content-type__save'));
+        expect(saveButton === null).toBe(false);
+        expect(saveButton.nativeElement.disabled).toBe(true);
+    });
+
+    it('should submit form when save button is clicked', () => {
+        const form: ContentTypesFormComponent = de.query(By.css('dot-content-types-form')).componentInstance;
+        spyOn(form, 'submitForm');
+        form.canSave = true;
+        fixture.detectChanges();
+
+        const saveButton = de.query(By.css('.content-type__save'));
+        saveButton.nativeElement.click();
+        expect(form.submitForm).toHaveBeenCalledTimes(1);
+    });
+
+    it('should bind save button disabled attribute to canSave property from the form', () => {
+        const form: ContentTypesFormComponent = de.query(By.css('dot-content-types-form')).componentInstance;
+        form.canSave = true;
+        fixture.detectChanges();
+
+        const saveButton = de.query(By.css('.content-type__save'));
+        expect(saveButton === null).toBe(false);
+        expect(saveButton.nativeElement.disabled).toBe(false);
+    });
+
+    it('should close the dialog and redirect', () => {
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(comp, 'cancelForm').and.callThrough();
+
+        const cancelButton = de.query(By.css('.content-type__cancel'));
+        cancelButton.nativeElement.click();
+        fixture.detectChanges();
+
+        expect(comp.cancelForm).toHaveBeenCalledTimes(1);
+        expect(comp.show).toBe(false);
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('/content-types-angular');
+    });
+
+    it('should NOT have dot-content-type-layout', () => {
         const contentTypeLayout = de.query(By.css('dot-content-type-layout'));
-        expect(contentTypeLayout).toBeTruthy();
+        expect(contentTypeLayout === null).toBe(true);
     });
 
-    it('should have Content Types Form', () => {
+    it('should have show form by default', () => {
+        const dialog = de.query(By.css('p-dialog'));
         const contentTypeForm = de.query(By.css('dot-content-types-form'));
-        expect(contentTypeForm).toBeTruthy();
+        expect(contentTypeForm === null).toBe(false);
+        expect(dialog === null).toBe(false);
     });
 
-    it('should NOT have Content Types Fields Drop Zone', () => {
-        const contentTypeForm = de.query(By.css('dot-content-type-fields-drop-zone'));
-        expect(contentTypeForm).toBeFalsy();
+    it('should have show form by default', () => {
+        const dialog = de.query(By.css('p-dialog'));
+        const contentTypeForm = de.query(By.css('dot-content-types-form'));
+
+        expect(comp.show).toBe(true);
+        expect(contentTypeForm === null).toBe(false);
+        expect(dialog === null).toBe(false);
     });
```

Comment:
the last two test have the same description

Reply:
It's duplicated, fixing it.